Cache/IO.lean

```diff
@@ -133,12 +133,23 @@ end
 
 def getPackageDirs : CacheM PackageDirs := return (← read).packageDirs
 
+/--
+`path` is assumed to be the unresolved file path corresping to a module:
+For `Mathlib.Init` this would be `Mathlib/Init.lean`.
+
+Find the root directory for `path`. This corresponds to the folder
+where the associated `.lake` folder lives, and usually is either `.`
```

Comment:
I don't think this is true. If I test it on `Verso`, the directory it returns is `workspace/verso/src/verso`, but the `.lake` folder is at `workspace/verso/.lake`. If you actually want the lake directory, you need to walk the ancestors until you find it.

A TODO comment about not supporting `srcDir` attributes in `lean_lib` would be acceptable for now.

Reply:
Or perhaps better a check that `.lake` exists, and erroring with a "not implemented" message when it doesn't.

Reply:
I've now renamed the function to `getSrcDir` because that's what it effectively does, and I've added the `TODO` and existence test of `.lake/` to `mkBuildPaths` beause that's where we assume no special options are set in `lake`.